### PR TITLE
WIP: How to decide which install method for OSX

### PR DESCRIPTION
**Pull request by gusseting**
This provides some basic instruction for how to find out if homebrew is
installed, for someone that may not be familiar with a terminal in any way.

### CODE REVIEW

**Comment by gusseting** [comment]
We weren't sure how to make a title link for the 2 install methods. Please change.

**Comment by tejasbubane** [comment]
If there are any aliases or functions for `brew`, `which` will return a different result:

```
⇒  alias brew=something
⇒  which brew
brew: aliased to something
```

This might give an impression that brew is not installed. 
I think using `brew --version` would be a better way to check than using `which brew`.

```
⇒  brew --version
Homebrew 0.9.9 (git revision 57bf; last commit 2016-05-09)
Homebrew/homebrew-core (git revision 1727; last commit 2016-05-09)
```


PS: I am using `zsh`, not sure how other shells behave with `which`.

**Comment by kotp** [comment]
You can avoid the alias problem by prefacing the command with a `\` as in `\brew --version` and that is good advice.

Other shells may provide `whence`.

**Comment by kytrinyx** [comment]
Thank you @gusseting - this is a sweet improvement to the onboarding process!

@kotp @tejasbubane thanks to both of you for jumping in here. Can I leave this topic/issue in your capable hands? If so: would you mind calling me over when it's ready to merge?

**Comment by kotp** [comment]
@gusserting is still waiting on an answer for the links.  Once she has that, she can hopefully edit the title from WIP and we can get this in.

Of course, we need to clarify the instructions for finding out if brew is installed, so are waiting for that as well, @gusseting is likely contemplating it at this moment.

File: x/docs/md/cli/mac.md [hunk 1]
```diff
@@ -1,3 +1,31 @@
+### Methods of Installing Exercism CLI
+
+  - Via [Homebrew](#title link)
+  - Via [Without Homebrew](#title Link2)
```
Comment (sj26) [inline x/docs/md/cli/mac.md]: These are just based on the title names, so:

```
  - Via [Homebrew](#installing-with-homebrew)
  - Via [Without Homebrew](#installing-without-homebrew)
```

**Comment by sj26** [comment]
Simply running `brew --version` should be enough to reveal if homebrew is installed. Output including "homebrew" and a version like:

```
$ brew --version
Homebrew 0.9.9 (git revision aae4; last commit 2016-05-13)
Homebrew/homebrew-core (git revision 72a7; last commit 2016-05-14)
```

Otherwise the shell, whatever it happens to be, should just report that the command isn't found:

```
$ brew
bash: brew: command not found
```

```
$ brew
zsh: command not found: brew
```

If there is a shell alias in place for brew then you'd hope that the user understands that alias and any effect it may have on this command, would be aware of having homebrew installed, or the alias would be sensible enough to pass through a `--version` flag call.

**Comment by gusseting** [comment]
@kotp @tejasbubane ok, I've made the changes that you've suggested, and added in links as per suggestion from @sj26 (thanks, Sam!). 
If there is anything further that I need to do before this pull request is merged, could you please @gusseting and make it very clear exactly what I need to do?
When comments are made, as I'm new to github + exercism and had already spent over 3 hours with @kotp on this pull request, I don't know what (if anything) I should be doing with new comments, which is why I hadn't made any changes.

File: x/docs/md/cli/mac.md [hunk 2]
```diff
@@ -1,3 +1,32 @@
...
+
+Homebrew is a package manager for OS X which installs the stuff that apple
+didn't.
+
+Find out if you have homebrew installed via the terminal.
+
+You can open a terminal using Spotlight by pressing <command><spacebar> and
+then type 'terminal' in the space provided.
+
+Type in the command:
```
Comment (tejasbubane) [inline x/docs/md/cli/mac.md]: Just to make sure this is not confused with the `Spotlight` thing above, replace this line with `On the command prompt, type in the command:`.

File: x/docs/md/cli/mac.md [hunk 3]
```diff
@@ -1,3 +1,32 @@
...
+
+`\brew --version`
+
+If homebrew is installed you may see output like the following:
```
Comment (tejasbubane) [inline x/docs/md/cli/mac.md]: The version numbers may vary so we should mention that as well. 
Something like `If homebrew is installed you may see output like the following (version numbers may vary):`

**Comment by tejasbubane** [comment]
Hey @gusseting sorry if our comments were a little vague. We really appreciate your work and we will try to be more specific henceforth. But if you get stuck or cannot figure out something and find yourself spending in more time than required, do not hesitate to comment on the issue asking for more clarification. We would be happy to help.

I have made a couple of small comments. After that I think we are good to go! :+1:

**Comment by tejasbubane** [comment]
Thanks @gusseting! this looks good :+1: cc:/ @kotp 

One last thing. You have 4 commits. Could you please `squash` those commits? You can follow this guide: https://github.com/edx/edx-platform/wiki/How-to-Rebase-a-Pull-Request#squash-your-changes And then remove the `WIP` from the heading.

**Comment by kotp** [comment]
If this looks good, we can squash them via the 'merge pull request' button here.  But it is a good thing to know how to do.  Let us know @gusseting what you want to do.

**Comment by gusseting** [comment]
@tejasbubane @kotp thanks for your feedback - I'll `squash` those as a learning exercise asap and will let you know when I'm done.

**Comment by gusseting** [comment]
@kotp @tejasbubane ok - i've tried to squash, but have done something wrong, as I'm getting errors from katrina's work (you have the error message on gitter, victor), and aborted. in the name of sanity, i think it's best if you could "merge pull request" on your end, and I can save the joy of `squash` for another day.

**Comment by tejasbubane** [comment]
No worries @gusseting.

@kytrinyx can you have a look at this and squash before merging?

**Comment by kotp** [comment]
@gusseting I accidentally clicked the merge without squashing.  Hate doing this on the browser.

Your work is now in the master branch.

**Comment by kytrinyx** [comment]
@kotp also reverted it. -- @kotp how do you want them to procede?

**Comment by kytrinyx** [comment]
Oh, never mind. The "revert" was a pull request that got deleted. So yes: :tada: your code is on master and will go live today.

Thank you :sparkles: :heart: :sparkles:

**Pull request merged**